da/celestia/mock/server.go

```diff
@@ -70,137 +100,135 @@ func (s *Server) Stop() {
-	if res.Code != da.StatusSuccess {
-		s.writeError(w, errors.New(res.Message))
-		return
-	}
-
-	var nShares []NamespacedShare
-	for _, block := range res.Blocks {
-		blob, err := block.MarshalBinary()
+	switch req.Method {
+	case "header.GetByHeight":
+		var params []interface{}
+		err := json.Unmarshal(req.Params, &params)
 		if err != nil {
 			s.writeError(w, err)
 			return
 		}
-		delimited, err := marshalDelimited(blob)
+		height := uint64(params[0].(float64))
+		dah := s.mock.GetHeaderByHeight(height)
+		resp := &response{
+			Jsonrpc: "2.0",
+			Result: header.ExtendedHeader{
+				DAH: dah,
+			},
+			ID:    req.ID,
+			Error: nil,
+		}
+		bytes, err := json.Marshal(resp)
 		if err != nil {
 			s.writeError(w, err)
+			return
 		}
-		nShares = appendToShares(nShares, []byte{1, 2, 3, 4, 5, 6, 7, 8}, delimited)
-	}
-	shares := make([]Share, len(nShares))
-	for i := range nShares {
-		shares[i] = nShares[i].Share
-	}
-
-	resp, err := json.Marshal(namespacedSharesResponse{
-		Shares: shares,
-		Height: res.DAHeight,
-	})
-	if err != nil {
-		s.writeError(w, err)
-		return
-	}
-
-	s.writeResponse(w, resp)
-}
-
-func (s *Server) data(w http.ResponseWriter, r *http.Request) {
-	height, err := parseHeight(r)
-	if err != nil {
-		s.writeError(w, err)
-		return
-	}
-
-	res := s.mock.RetrieveBlocks(r.Context(), height)
-	if res.Code != da.StatusSuccess {
-		s.writeError(w, errors.New(res.Message))
-		return
-	}
-
-	data := make([][]byte, len(res.Blocks))
-	for i := range res.Blocks {
-		data[i], err = res.Blocks[i].MarshalBinary()
+		s.writeResponse(w, bytes)
+	case "blob.GetAll":
+		var params []interface{}
+		err := json.Unmarshal(req.Params, &params)
+		if err != nil {
+			s.writeError(w, err)
+			return
+		}
+		height := params[0].(float64)
+		nsBase64 := params[1].([]interface{})[0].(string)
+		ns, err := base64.StdEncoding.DecodeString(nsBase64)
+		if err != nil {
+			s.writeError(w, err)
+			return
+		}
+		block := s.mock.RetrieveBlocks(r.Context(), uint64(height))
+		var blobs []blob.Blob
+		for _, block := range block.Blocks {
+			data, err := block.MarshalBinary()
+			if err != nil {
+				s.writeError(w, err)
+				return
+			}
+			blob, err := blob.NewBlobV0(ns, data)
+			if err != nil {
+				s.writeError(w, err)
+				return
+			}
+			blobs = append(blobs, *blob)
+		}
+		resp := &response{
+			Jsonrpc: "2.0",
+			Result:  blobs,
+			ID:      req.ID,
+			Error:   nil,
+		}
+		bytes, err := json.Marshal(resp)
+		if err != nil {
+			s.writeError(w, err)
+			return
+		}
+		s.writeResponse(w, bytes)
+	case "share.SharesAvailable":
+		resp := &response{
+			Jsonrpc: "2.0",
+			ID:      req.ID,
+			Error:   nil,
+		}
+		bytes, err := json.Marshal(resp)
+		if err != nil {
+			s.writeError(w, err)
+			return
+		}
+		s.writeResponse(w, bytes)
+	case "state.SubmitPayForBlob":
+		var params []interface{}
+		err := json.Unmarshal(req.Params, &params)
+		if err != nil {
+			s.writeError(w, err)
+			return
+		}
+		if len(params) != 3 {
```

Comment:
Follow up, we should add a comment here about what the 3 params should be for context.